hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/ObjectBuilder.kt

```diff
@@ -0,0 +1,215 @@
+      )
+    },
+    aliases = this.aliases.map { a ->
+      Alias(
+        gender = a.gender,
+        ethnicity = a.ethnicity,
+        firstName = this.firstName,
+        middleName = null,
+        lastName = this.lastName,
+        age = null,
+        dob = LocalDate.parse(this.dateOfBirth),
+        nameType = null,
+        createDate = LocalDate.now(),
+      )
+    },
+    physicalCharacteristics = mutableListOf<PhysicalCharacteristic>().also { pcs ->
+      this.physicalCharacteristics?.hairColour?.let {
+        pcs.add(PhysicalCharacteristic("HAIR", "Hair Colour", it, null))
+      }
+      this.physicalCharacteristics?.rightEyeColour?.let {
+        pcs.add(PhysicalCharacteristic("R_EYE_C", "Right Eye Colour", it, null))
+      }
+      this.physicalCharacteristics?.leftEyeColour?.let {
+        pcs.add(PhysicalCharacteristic("L_EYE_C", "Left Eye Colour", it, null))
+      }
+      this.physicalCharacteristics?.facialHair?.let {
+        pcs.add(PhysicalCharacteristic("FACIAL_HAIR", "Facial Hair", it, null))
+      }
+      this.physicalCharacteristics?.shapeOfFace?.let {
+        pcs.add(PhysicalCharacteristic("FACE", "Shape of Face", it, null))
+      }
+      this.physicalCharacteristics?.build?.let {
+        pcs.add(PhysicalCharacteristic("BUILD", "Build", it, null))
+      }
+      this.physicalCharacteristics?.shoeSize?.let {
+        pcs.add(PhysicalCharacteristic("SHOESIZE", "Shoe Size", it.toString(), null))
+      }
+    },
+    physicalMarks = mutableListOf<PhysicalMark>().also { pms ->
+      this.physicalMarks?.tattoo?.forEach {
+        pms.add(PhysicalMark("Tattoo", null, it.bodyPart, null, it.comment, null))
+      }
+      this.physicalMarks?.mark?.forEach {
+        pms.add(PhysicalMark("Mark", null, it.bodyPart, null, it.comment, null))
+      }
+      this.physicalMarks?.other?.forEach {
+        pms.add(PhysicalMark("Other", null, it.bodyPart, null, it.comment, null))
+      }
+      this.physicalMarks?.scar?.forEach {
+        pms.add(PhysicalMark("Scar", null, it.bodyPart, null, it.comment, null))
+      }
+    },
+  ).let {
+    if (released) {
+      it.copy(
+        status = "INACTIVE OUT",
+        lastMovementTypeCode = "REL",
+        lastMovementReasonCode = "HP",
+        inOutStatus = "OUT",
+        agencyId = "OUT",
+      )
+    } else {
+      it.copy(
+        lastMovementTypeCode = "ADM",
+        lastMovementReasonCode = "I",
+      )
+    }
+  }
+
+fun PrisonerBuilder.toPrisoner(): Prisoner =
+  Prisoner(ob = toOffenderBooking(), incentiveLevel = toIncentiveLevel(), null)
+
+private fun getOffenderBookingTemplate(): OffenderBooking =
+  GsonConfig().gson().fromJson("/templates/booking.json".readResourceAsText(), OffenderBooking::class.java)
+
+fun Prisoner(ob: OffenderBooking, incentiveLevel: IncentiveLevelDto?, restrictedPatientData: RestrictivePatient?) =
+  Prisoner().apply { this.translate(null, ob, Result.success(incentiveLevel), restrictedPatientData) }
```

Comment:
Is the plan to get rid of OffenderBooking at some pont? The search doesn't really need to know about it at all once you've unpicked all the indexing test set up.

Reply:
yes - hoping to get that all tidied up once all code is in and working